sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/e-cert-file-handler.ts

```diff
@@ -99,97 +108,123 @@ export class ECertFileHandler extends ESDCFileHandler {
+   * @param sequenceNumber e-Cert sequence number.
+   * @param entityManager manages the current DB transaction.
+   * @param eCertIntegrationService Full-Time/Part-Time integration responsible
+   * for the respective integration.
+   * @param offeringIntensity disbursement offering intensity.
+   * @param fileCode file code applicable for Part-Time or Full-Time.
+   * @returns information of the uploaded e-Cert file.
+   * @throws CustomNamedError ECERT_GENERATION_NO_RECORDS_AVAILABLE
+   */
+  private async processECert(
+    sequenceNumber: number,
+    entityManager: EntityManager,
+    eCertIntegrationService: ECertIntegrationService,
+    offeringIntensity: OfferingIntensity,
+    fileCode: string,
+  ): Promise<ECertUploadResult> {
+    try {
+      const disbursements =
+        await this.eCertGenerationService.prepareDisbursementsForECertGeneration(
+          offeringIntensity,
+          entityManager,
+        );
+      if (!disbursements.length) {
+        throw new CustomNamedError(
+          `There are no records available to generate an e-Cert file for ${offeringIntensity}`,
+          ECERT_GENERATION_NO_RECORDS_AVAILABLE,
+        );
+      }
+      this.logger.log(
+        `Found ${disbursements.length} ${offeringIntensity} disbursements schedules.`,
+      );
+      const disbursementRecords = disbursements.map((disbursement) => {
+        return this.createECertRecord(disbursement);
+      });
 
-          this.logger.log(`Uploading ${offeringIntensity} content...`);
-          await eCertIntegrationService.uploadContent(
-            fileContent,
-            fileInfo.filePath,
-          );
+      this.logger.log(`Creating  ${offeringIntensity} e-Cert file content...`);
+      const fileContent = eCertIntegrationService.createRequestContent(
+        disbursementRecords,
+        sequenceNumber,
+      );
 
-          uploadResult = {
-            generatedFile: fileInfo.filePath,
-            uploadedRecords: disbursementRecords.length,
-          };
-        } catch (error) {
-          this.logger.error(
-            `Error while uploading content for ${offeringIntensity} e-Cert file: ${error}`,
-          );
-          throw error;
-        }
-      },
-    );
-    return uploadResult;
+      // Create the request filename with the file path for the e-Cert File.
+      const fileInfo = this.createRequestFileName(fileCode, sequenceNumber);
+      this.logger.log(`Uploading ${offeringIntensity} content...`);
+      await eCertIntegrationService.uploadContent(
+        fileContent,
+        fileInfo.filePath,
+      );
+      return {
+        generatedFile: fileInfo.filePath,
+        uploadedRecords: disbursementRecords.length,
+      };
+    } catch (error: unknown) {
+      if (
+        error instanceof CustomNamedError &&
+        error.name !== ECERT_GENERATION_NO_RECORDS_AVAILABLE
+      ) {
+        this.logger.error(
+          `Error while uploading content for ${offeringIntensity} e-Cert file: ${error}`,
```

Comment:
if there is an error after line 182. then will we rollback the logic executed inside 182

Reply:
Any error inside `processECert` method will rollback the entire transaction. For instance, if the file is not possible to be uploaded due to some SFTP connection error this will revert the entire DB changes and e-Cert generation.